# 대응 표본 T-검정 (Paired T-test)
표본의 각 사례마다 대응하는 2개의 관측치를 통해 판단

한 집단에 어떤 작용이 가해졌을 때에 대한 before, after를 비교
## #01. 작업준비

### 패키지 참조


In [2]:
from scipy import stats
import pandas as pd
from matplotlib import pyplot as plt
import seaborn as sb

## #02. 예제(1) - 사슴의 다리 길이 비교
사슴의 왼쪽 뒷다리와 왼쪽 앞다리의 길이를 측정한 자료를 통해 두 다리의 길이(cm)가 같은지 다른지 알고 싶다.

이 경우 한 사슴에 대하여 두 다리 길이를 측정하였으므로 사슴은 독립이지만 한 마리의 사슴의 일부인 두 다리는 독 립이 아니다.

In [3]:
df = pd.read_excel("https://data.hossam.kr/E02/deer.xlsx", index_col="사슴")
df

뒷다리  앞다리
사슴          
1   142  138
2   140  136
3   144  147
4   144  139
5   142  143
6   146  141
7   149  143
8   150  145
9   142  136
10  148  146

## T-Test 시행
### case1 


귀무가설($H_0$)  : 사슴의 앞다리와 뒷다리 길이에는 차이가 없다

대립가설($H_1$)  :사슴의 앞다리와 뒷다리 길이에는 차이가 있다

전 후의 평균의 차이가있는지 검정


In [4]:
stats.ttest_rel(df['뒷다리'],df['앞다리'],alternative="two-sided")

TtestResult(statistic=3.413793103448276, pvalue=0.007703223347263766, df=9)



### 결과 해석

p값이 0.05보다 작으므로 귀무가설 기각, 유의수준 0.05에서 앞다리와 뒷다리의 길이에는 차이가 있다.

### case2 
귀무가설($H_0$)  : 사슴의 앞다리와 뒷다리 길이에는 길지않다

대립가설($H_1$)  :사슴의 앞다리와 뒷다리 길이에는 길다

In [6]:
stats.ttest_rel(df['앞다리'],df['뒷다리'],alternative='greater')


TtestResult(statistic=-3.413793103448276, pvalue=0.9961483883263681, df=9)

#### 결과해석

T-통계량은 -3.413, p값 0.996148인 것을 확인.  유의수준 0.05에서 p값이 0.05보다 크므로 귀무가설을 채택. 앞다리가 뒷다리보다 길지 않다고 할 수 있음.

### case3


귀무가설($H_0$)  : 사슴의 앞다리는 뒷다리보다 짧지 않다

대립가설($H_1$)  : 사슴의 앞다리는 뒷다리보다 짧다

In [7]:
stats.ttest_rel(df['앞다리'],df['뒷다리'],alternative='less')


TtestResult(statistic=-3.413793103448276, pvalue=0.003851611673631883, df=9)

#### 결과해석
차이값에 대한 t통계량은 -3.413, 대응되는 유의확률이 0.0038로 0.05보다 작다. 따라서 사슴의 앞다리는 뒷다리보다 유의수준 0.05에서 짧다고 할수잇다 .


## #03. 예제(2) - 자동차 첨가제가 주행거리에 미치는 영향
다섯 종류의 새 차에 대하여 같은 종류의 차 두 대 중에서 한 대를 랜덤하게 선택하여 첨가제를 사용하고, 다른 한 대에는 첨가제를 사용하지 않고 같은 운전자가 같은 장소에서 운전한 결과 다음과 같은 자료를 얻었다.

첨가제를 사용하면 주행거리가 늘어난다고 할 수 있는지 유의수준 5%에서 검정하여 보라.

그룹A : 첨가제를 사용한 경우, 그룹B : 첨가제를 사용하지 않은 경우
### 데이터 가져오기

In [8]:
df = pd.read_excel("https://data.hossam.kr/E02/oil.xlsx")
df

그룹A   그룹B
0  11.8  11.4
1  13.9  13.1
2  16.3  16.1
3  11.6  10.9
4   8.4   8.3

### case1

귀무가설($H_0$)  : 그룹A와 그룹B의 주행거리에는 차이가 없다. 실험에 사용된 첨가제는 주행거리에 영향을 주지 않는다.

대립가설($H_1$)  :  그룹A와 그룹B의 주행거리에는 차이가 있다. 실험에 사용된 첨가제는 주행거리에 영향을 준다.

In [9]:
stats.ttest_rel(df['그룹A'],df['그룹B'],alternative='two-sided')

TtestResult(statistic=3.226236530469768, pvalue=0.03208991550156059, df=4)

#### 결과해석 
대응표본 t검정 결과에서 검정통계량 값이 3.22이고 유의확률이 0.032인것을 확인했고 유의확률이 0.05보다 작으므로 귀무가설을 기각하고 대립가설을 채택한다. 즉, A와 B는 주행거리에 차이가 있다고 판한됨.

### case 2


귀무가설($H_0$) : 그룹A의 주행거리는 그룹B의 주행거리보다 짧지 않다.


대립가설($H_1$) :그룹A의 주행거리는 그룹B의 주행거리보다 짧다.


In [10]:
stats.ttest_rel(df['그룹A'],df['그룹B'],alternative="less")

TtestResult(statistic=3.226236530469768, pvalue=0.9839550422492197, df=4)

#### 결과해석 
p값이 0.05보다 크므로 귀무가설 채택.  첨가제를 넣은 그룹의 주행거리가 첨가제를 넣지않은 그룹의 주행거리보다 짧지않다

### case 3

귀무가설($H_0$) : 그룹A의 주행거리는 그룹B의 주행거리보다 길지 않다.


대립가설($H_1$) :그룹A의 주행거리는 그룹B의 주행거리보다 길다.


In [11]:
stats.ttest_rel(df['그룹A'],df['그룹B'],alternative="greater")

TtestResult(statistic=3.226236530469768, pvalue=0.016044957750780294, df=4)

#### 결과해석 
p값이 0.05보다 작으므로 대립가설을 채택. 즉 첨가제를 넣은 그룹의 주행거리가 첨가제를 넣지않은 그룹의 주해거리보다 길다는 것을 확인할 수 있ㅇ다. 